hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -557,6 +567,19 @@ private Object invoke(String nsId, int retryCount, final Method method,
           }
 
           // retry
+          try {
```

Comment:
The idea before was for us to surface the unavailability to the client and then the client retry with the regular client logic.

Reply:
In fact, not all clients are so smart, they may not configred appropriately. When those clients encounter this problem, they complain about router not retry more.

Reply:
Sure, but we should change the logic to match the old style or at least to not have the two styles together.